### PR TITLE
RASS-69 add tables via flyway

### PR DESCRIPTION
**Pull request by carloveo-moj**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/jpa/entity/AppearanceOutcomeEntity.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import jakarta.persistence.Id
+import jakarta.persistence.Table
+
+@Entity
+@Table(name = "appearance_outcome")
+data class AppearanceOutcomeEntity(
+  @Id
+  @Column
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  val id: Int? = null,
```
Comment (iterative-it) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsremandandsentencingapi/jpa/entity/AppearanceOutcomeEntity.kt:16]: Will using nullable ids cause any unnecessary null assertions down the line? Using `val id: Int = 0` will achieve the same JPA behaviour I believe.

File: docker-compose.yml [hunk 2]
```diff
@@ -12,6 +12,15 @@ services:
...
       - SPRING_PROFILES_ACTIVE=dev
-
+  remand-and-sentencing-db:
+    image: postgres:15
+    networks:
+      - hmpps
+    ports:
+      - "5432:5432"
+    environment:
+      - POSTGRES_PASSWORD=ras-pass
```
Comment (iterative-it) [inline docker-compose.yml:22]: If you ever end up using the `java_localstack_postgres` executor on CircleCI, the db, username and password have to be the same thing. Just a heads up based on something that tripped me up :)

File: src/main/resources/migration/common/V1__remand_tables.sql [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    charge_outcome_id int references charge_outcome(id) NOT NULL,
+    superseding_charge_id int references charge(id)
+);
+
+CREATE TABLE appearance_charge(
+    id SERIAL PRIMARY KEY,
+    appearance_id int references court_appearance(id) NOT NULL,
+    charge_id int references charge(id) NOT NULL
+);
+
```
Comment (iterative-it) [inline src/main/resources/migration/common/V1__remand_tables.sql:64]: Possibly some indexes missing here but they can all be added later

**Review by iterative-it — DISMISSED** [review]
LGTM - couple of minor comments

File: src/main/resources/migration/common/V1__remand_tables.sql (hunk 3 above)
Comment (carloveo-moj) [reply src/main/resources/migration/common/V1__remand_tables.sql:64]: not entirely sure how the data will be retrieved yet, lots to think about

**Review by iterative-it — APPROVED** [review]
LGTM

**Pull request merged**